heartbeat/monitors/active/http/respbody.go

```diff
@@ -28,30 +28,26 @@ import (
 
 	"github.com/elastic/beats/v7/heartbeat/reason"
 	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/mime"
 )
 
 // maxBufferBodyBytes sets a hard limit on how much we're willing to buffer for any reason internally.
 // since we must buffer the whole body for body validators this is effectively a cap on that.
 // 100MiB out to be enough for everybody.
 const maxBufferBodyBytes = 100 * units.MiB
 
-func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, reason.Reason) {
+func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, string, reason.Reason) {
 	// Determine how much of the body to actually buffer in memory
-	var bufferBodyBytes int
-	if validator.wantsBody() {
-		bufferBodyBytes = maxBufferBodyBytes
-	} else if config.IncludeBody == "always" || config.IncludeBody == "on_error" {
+	bufferBodyBytes := maxBufferBodyBytes
+	if !validator.wantsBody() && (config.IncludeBody == "always" || config.IncludeBody == "on_error") {
 		// If the user has asked for bodies to be recorded we only need to buffer that much
 		bufferBodyBytes = config.IncludeBodyMaxBytes
```

Comment:
The conditional above keeps the same behavior as prior -- if `validator.wantsBody()` is `true` or if we have no `config.IncludeBodyMaxBytes` specified, then we just use `maxBufferBodyBytes` -- that way we can still use `config.IncludeBodyMaxBytes` as a mechanism for limiting the amount of bytes we read off the wire.

Alternatively, maybe we can just use `config.IncludeBodyMaxBytes` if it's non-0 and use `maxBufferBodyBytes` otherwise?